support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoAuthenticationMetaDataPopulator.java

```diff
@@ -0,0 +1,38 @@
+package org.apereo.cas.adaptors.duo.authn;
+
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.cas.authentication.AuthenticationBuilder;
+import org.apereo.cas.authentication.AuthenticationManager;
+import org.apereo.cas.authentication.AuthenticationTransaction;
+import org.apereo.cas.authentication.Credential;
+import org.apereo.cas.authentication.metadata.BaseAuthenticationMetaDataPopulator;
+
+/**
+ * Implementation of AuthenticationMetadaPopulator to handle multiple Duo instances.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+@Slf4j
+@ToString(callSuper = true)
+@RequiredArgsConstructor
+public class DuoAuthenticationMetaDataPopulator extends BaseAuthenticationMetaDataPopulator {
+    private final String authenticationContextAttribute;
+
+    private final DuoAuthenticationHandler authenticationHandler;
+
```

Comment:
Use the id of the specific provider as credential metadata so it can be matched correctly in ContextValidator.